internal/services/manageddevopspools/managed_devops_pool_resource.go

```diff
@@ -140,13 +159,57 @@ func (r ManagedDevOpsPoolResource) Update() sdk.ResourceFunc {
 				return fmt.Errorf("retrieving %s: `properties` was nil", id)
 			}
 
-			properties := existing.Model
+			payload := existing.Model
+
+			if metadata.ResourceData.HasChange("identity") {
+				identity, err := identity.ExpandLegacySystemAndUserAssignedMapFromModel(config.Identity)
+				if err != nil {
+					return fmt.Errorf("expanding `identity`: %+v", err)
+				}
+				payload.Identity = identity
+			}
+
+			if metadata.ResourceData.HasChange("dev_center_project_resource_id") {
+				payload.Properties.DevCenterProjectResourceId = config.DevCenterProjectResourceId
+			}
+
+			if metadata.ResourceData.HasChange("maximum_concurrency") {
+				payload.Properties.MaximumConcurrency = config.MaximumConcurrency
+			}
+
+			if metadata.ResourceData.HasChange("location") {
```

Comment:
`Location` is a `ForceNew` property which will not enter `Update` method. Please double check the code and remove all `ForceNew` properties from `Update`

Reply:
I see, thanks for sharing. Have updated and cross checked all ForceNew properties (location and ResourceGroup) and now none of them are being used in Update flow.